File: app/schemas/gov_meeting.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, List


# ── Enterprise-side requests ──────────────────────────────────────────────────

class GovMeetingSubmitRequest(BaseModel):
    contactName: str = Field(..., max_length=100)
    contactPhone: str = Field(..., max_length=64)
    topicCode: Optional[str] = None
    topicName: Optional[str] = None
    meetingLevel: Optional[str] = None
    expectedLevelCode: Optional[str] = None
    expectedLevelName: Optional[str] = None
    title: Optional[str] = Field(None, max_length=300)
    content: Optional[str] = None          # 映射到 meeting_content
    discussionItem: Optional[str] = None
    urgencyLevel: Optional[str] = None
    industryCode: Optional[str] = None
    industryName: Optional[str] = None
    registeredAddress: Optional[str] = None
    description: Optional[str] = None     # 兼容旧字段（申请说明）
    commitmentChecked: int = Field(..., ge=1, le=1, description="必须为1")
    regionCode: str
    regionName: str
    serviceCenterId: Optional[int] = None
    serviceCenterName: Optional[str] = None
    attachmentIds: Optional[List[int]] = None


class GovMeetingModifyRequest(BaseModel):
    contactName: Optional[str] = Field(None, max_length=100)
    contactPhone: Optional[str] = Field(None, max_length=64)
    topicCode: Optional[str] = None
    topicName: Optional[str] = None
    meetingLevel: Optional[str] = None
    expectedLevelCode: Optional[str] = None
    expectedLevelName: Optional[str] = None
    title: Optional[str] = None
    content: Optional[str] = None
    discussionItem: Optional[str] = None
    urgencyLevel: Optional[str] = None
    description: Optional[str] = None
    attachmentIds: Optional[List[int]] = None


class GovMeetingSupplementRequest(BaseModel):
    content: Optional[str] = None          # 补充说明内容
    description: Optional[str] = None      # 兼容旧字段
    attachmentIds: Optional[List[int]] = None


class GovMeetingEvaluationRequest(BaseModel):
    satisfaction: str
    score: int = Field(..., ge=1, le=5)
    resolvedFlag: Optional[int] = None
    comment: Optional[str] = Field(None, max_length=1000)


# ── Admin-side requests ───────────────────────────────────────────────────────

class GovMeetingAuditRequest(BaseModel):
    auditType: str  # ACCEPT / REJECT / RETURN_SUPPLEMENT
    rejectReasonCode: Optional[str] = None
    rejectReasonName: Optional[str] = None
    finalLevelCode: Optional[str] = None   # 后台研判约见层级（ACCEPT 时填写）
    finalLevelName: Optional[str] = None
    opinion: Optional[str] = None


class ParticipantItem(BaseModel):
    participantType: str   # GOV / ENTERPRISE
    participantName: str
    participantTitle: Optional[str] = None
    participantDeptId: Optional[str] = None
    participantDeptName: Optional[str] = None
    contactPhone: Optional[str] = None
    roleName: Optional[str] = None
    sortNo: Optional[int] = 0


class GovMeetingArrangeRequest(BaseModel):
    meetingDate: Optional[datetime] = None
    meetingPlace: Optional[str] = None
    meetingMethod: Optional[str] = None
    startTime: Optional[datetime] = None
    endTime: Optional[datetime] = None
    hostDeptId: Optional[str] = None
    hostDeptName: Optional[str] = None
    notes: Optional[str] = None
    govContactName: Optional[str] = None
    govContactPhone: Optional[str] = None
    remark: Optional[str] = None
    participants: Optional[List[ParticipantItem]] = None


class GovMeetingRecordRequest(BaseModel):
    arrangementId: Optional[int] = None
    content: str
    conclusions: Optional[str] = None
    followUpItems: Optional[str] = None
    recordTime: Optional[datetime] = None


class GovMeetingConfirmRequest(BaseModel):
    opinion: Optional[str] = None


class GovMeetingCompleteRequest(BaseModel):
    meetingAt: Optional[datetime] = None
    opinion: Optional[str] = None


class GovMeetingFinishRequest(BaseModel):
    sendEvaluation: Optional[int] = Field(1, description="1=发送评价通知 0=直接办结")
    opinion: Optional[str] = None

File: admin-web/src/api/govMeeting.ts
import { get, post, put } from './request'
import type { PageResult } from './appeal'

// ── Interfaces ────────────────────────────────────────────────────────────────

export interface GovMeetingApply {
  id: number
  applyNo: string
  enterpriseId: number
  enterpriseName: string
  creditCode: string
  contactName: string
  contactPhone: string
  topicCode?: string
  topicName?: string
  meetingLevel?: string
  expectedLevelCode?: string
  expectedLevelName?: string
  finalLevelCode?: string
  finalLevelName?: string
  description: string
  commitmentChecked?: number
  regionCode: string
  regionName: string
  serviceCenterId?: number
  serviceCenterName?: string
  status: string
  rejectReasonCode?: string
  rejectReasonName?: string
  rejectOpinion?: string
  submittedAt?: string
  acceptedAt?: string
  arrangedAt?: string
  meetingAt?: string
  completedAt?: string
  evaluatedAt?: string
  finishedAt?: string
  createdAt?: string
}

export interface GovMeetingParticipant {
  id?: number
  participantType: string   // GOV / ENTERPRISE
  participantName: string
  participantTitle?: string
  participantDeptId?: string
  participantDeptName?: string
  contactPhone?: string
  roleName?: string
  sortNo?: number
}

export interface GovMeetingArrangement {
  id: number
  applyId: number
  meetingDate?: string
  meetingPlace?: string
  meetingMethod?: string
  startTime?: string
  endTime?: string
  hostDeptId?: string
  hostDeptName?: string
  notes?: string
  govContactName?: string
  govContactPhone?: string
  remark?: string
  confirmedFlag: number
  participants: GovMeetingParticipant[]
}

export interface GovMeetingRecord {
  id: number
  applyId: number
  arrangementId?: number
  content: string
  conclusions?: string
  followUpItems?: string
  recorderName?: string
  recordTime?: string
  createdAt?: string
}

export interface GovMeetingAudit {
  id: number
  applyId: number
  actionType: string
  actionName: string
  beforeStatus?: string
  afterStatus?: string
  opinion?: string
  operatorType?: string
  operatorId?: string
  operatorName?: string
  operatorDeptId?: string
  operatorDeptName?: string
  createdAt: string
}

export interface GovMeetingAttachment {
  id: number
  originalName: string
  fileExt?: string
  fileSize?: number
  storagePath?: string
}

export interface GovMeetingEvaluation {
  id: number
  satisfaction: string
  score: number
  resolvedFlag?: number
  comment?: string
  createdAt?: string
}

export interface GovMeetingDetail extends GovMeetingApply {
  arrangement?: GovMeetingArrangement | null
  auditTrail: GovMeetingAudit[]
  records: GovMeetingRecord[]
  attachments: GovMeetingAttachment[]
  evaluation?: GovMeetingEvaluation | null
}

export interface GovMeetingListParams {
  enterpriseName?: string
  creditCode?: string
  status?: string
  topicCode?: string
  regionCode?: string
  serviceCenterId?: number
  startDate?: string
  endDate?: string
  pageNo?: number
  pageSize?: number
}

// ── API functions ─────────────────────────────────────────────────────────────

export function getGovMeetingList(params: GovMeetingListParams): Promise<PageResult<GovMeetingApply>> {
  return get<PageResult<GovMeetingApply>>('/api/admin/gov-meetings', params as Record<string, unknown>)
}

export function getGovMeetingDetail(id: number): Promise<GovMeetingDetail> {
  return get<GovMeetingDetail>(`/api/admin/gov-meetings/${id}`)
}

/** 审核：auditType = ACCEPT | RETURN_SUPPLEMENT | REJECT */
export function auditGovMeeting(id: number, data: {
  auditType: 'ACCEPT' | 'RETURN_SUPPLEMENT' | 'REJECT'
  opinion?: string
  rejectReasonCode?: string
  rejectReasonName?: string
  finalLevelCode?: string
  finalLevelName?: string
}): Promise<GovMeetingApply> {
  return post<GovMeetingApply>(`/api/admin/gov-meetings/${id}/audit`, data)
}

/** 安排约见（PENDING_ARRANGE → ARRANGED） */
export function arrangeGovMeeting(id: number, data: {
  meetingDate?: string
  meetingPlace?: string
  meetingMethod?: string
  startTime?: string
  endTime?: string
  hostDeptId?: string
  hostDeptName?: string
  notes?: string
  govContactName?: string
  govContactPhone?: string
  remark?: string
  participants?: GovMeetingParticipant[]
}): Promise<GovMeetingArrangement> {
  return post<GovMeetingArrangement>(`/api/admin/gov-meetings/${id}/arrange`, data)
}

/** 修改约见安排（ARRANGED / WAIT_MEETING，不变状态） */
export function updateGovMeetingArrangement(id: number, arrId: number, data: {
  meetingDate?: string
  meetingPlace?: string
  meetingMethod?: string
  startTime?: string
  endTime?: string
  hostDeptId?: string
  hostDeptName?: string
  notes?: string
  govContactName?: string
  govContactPhone?: string
  remark?: string
  participants?: GovMeetingParticipant[]
}): Promise<GovMeetingArrangement> {
  return put<GovMeetingArrangement>(`/api/admin/gov-meetings/${id}/arrangements/${arrId}`, data)
}

/** 确认并通知企业（ARRANGED → WAIT_MEETING） */
export function confirmGovMeeting(id: number, data: { opinion?: string }): Promise<GovMeetingApply> {
  return post<GovMeetingApply>(`/api/admin/gov-meetings/${id}/confirm`, data)
}

/** 标记约见完成（WAIT_MEETING → MEETING_COMPLETED） */
export function completeGovMeeting(id: number, data: { meetingAt?: string; opinion?: string }): Promise<GovMeetingApply> {
  return post<GovMeetingApply>(`/api/admin/gov-meetings/${id}/complete`, data)
}

/** 填写约见纪要（不改变状态） */
export function recordGovMeeting(id: number, data: {
  arrangementId?: number
  content: string
  conclusions?: string
  followUpItems?: string
  recordTime?: string
}): Promise<GovMeetingRecord> {
  return post<GovMeetingRecord>(`/api/admin/gov-meetings/${id}/record`, data)
}

/** 办结：sendEvaluation=1 → PENDING_EVALUATION；sendEvaluation=0 → COMPLETED */
export function finishGovMeeting(id: number, data: { sendEvaluation?: number; opinion?: string }): Promise<GovMeetingApply> {
  return post<GovMeetingApply>(`/api/admin/gov-meetings/${id}/finish`, data)
}
